### DISCUSSION TITLE
Adamic Adar Link Prediction

### DISCUSSION THREAD

**Discussion by CathyWilms** [SkepticMystic/graph-analysis#69] [Q&A]
Hi, quick question. If I understand it right, the Academic Adar Analysis tells me what nodes **should be linked**. Thats really nice and useful. But the view is showing me noded that **already are linked** as well. It would be usefull to hide those to have a "for your consideration" view.
Do I missunderstand something? Im new to graph analysis.

![image](https://github.com/SkepticMystic/graph-analysis/assets/123727377/d3a64c9b-9fd0-4ad9-9136-3d10ac217c1a)

**Comment by SkepticMystic** [comment]
Hey Cathy, thanks for your question.
iirc, the little "chain link" icon indicates if two notes are already linked. I get your point though, maybe you should be able to filter those out, or move them to a different section. That isn't possible right now, but you can open a PFR for future.

I hope that helps

**Reply by hittjwX** [reply]
When working with a long document, it is good to see what is already linked. If this goes in, it needs to be in preference. I want to see both.

**Comment by hittjwX** [comment]
Taking a second look at this question, Adamic Adar Analysis (Adar Index) assists in referenced and unreferenced navigation.

It tells you what is in the subject domain worth considering as you move through this topic.

If you have the Adar Index, you are already linking to it in context. That context is the amount of information shared between two nodes. 

Any wiki link added to the note is a duplicate. Technically, the Adar Index lets you navigate between notes without a wiki link at all.

Some may want to link to get the Graph view. However, if you are doing research, does suggested contextually related content really earn its place in THIS note?

Since the Adar Index and Graph view can both be seen on the same screen, the suggested change can be vetted before it joins a note—if it is necessary at all.

It's not telling you what should be linked; it's telling you what might be potential connections by considering common neighbors. 

In research, you decide whether the relationship is worth pursuing. In that, vetting existing links too. The implementation here is pretty accurate.

In the following screenshot:

<img width="1600" height="860" alt="image" src="https://github.com/user-attachments/assets/1dc8b8b9-8495-4c3b-8f8e-de16915dba79" />

The Adar Index shows that the note "Experts to Interview" is a better match than the related links I previously used. Otherwise, it would scored the links I had higher.

Of the top notes, only the first is relevant because I was writing a goat expert about interviewing them for my overlapping-topic podcast. 

If this were research, I'd strengthen this topical community by removing weak links, inserting stronger ones, and then evaluating both notes to better understand the topic.

**Reply by hittjwX** [reply]
Clarification after testing in Obsidian:

- A note with very few or zero links will not return anything meaningful for the `Adamic Adar` tab. With zero links, it won't show anything until some other note links to it. If there is no network, or a weak one, there is nothing for the algorithm to do.
- Poor quality wiki links in notes add noise to the analysis. Curating the links improves the contextual (body of the note) relevance, telegraphed in the relationship (network nodal context). In my reply above, I'm not clear which context I'm referring to.
- Orphan notes don't get picked up in Adamic Adar; you'll need to use Jaccard to find similarity within the body of the note. Jaccard can offer link suggestions. Adamic Adar is context in the network, Jaccard is context in the note. 

When I'm taking notes, I often link back to the project file associated with research. Or take a guess about related notes. 

I put this in a `## Related` section. With curation, these links help with navigation and clustering, and they add some meaning -- otherwise surplus wiki links become a distraction.

I'm finding that fewer, more relevant links are helping my workflow. Yet no links may be leaving good notes hidden from view.

If you have a lot of notes and links, it takes Graph Analysis a few minutes to populate relationships. Spamming the "Refresh Index" button doesn't do anything.
